include/nbl/asset/IDescriptorSet.h

```diff
@@ -92,31 +93,31 @@ class IDescriptorSet : public virtual core::IReferenceCounted // TODO: try to re
 				}
 				~SDescriptorInfo()
 				{
-					if (desc && desc->getTypeCategory()==IDescriptor::EC_IMAGE)
+					if (desc and (desc->getTypeCategory()==IDescriptor::EC_IMAGE or desc->getTypeCategory() == IDescriptor::EC_SAMPLER))
```

Comment:
no when you are `EC_SAMPLER` then your `desc` is the sampler, and `sampler` is not supposed to be used

Reply:
see https://github.com/Devsh-Graphics-Programming/Nabla/pull/693/files#r1648876515